### ISSUE TITLE
[UX] CKEditor: Provide a "File" button, to match the functionality of the current "Image" button

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#6553]
This came up during my research for a better UX/workflow for #6552...

In Drupal, there is no "File uploading" option OOTB for when you are configuring text formats like we have in Backdrop, and some articles like https://www.drupal.org/forum/support/post-installation/2015-04-21/drupal-8-what-is-the-best-way-to-add-file-upload-to suggest that you need contributed modules like https://www.drupal.org/project/imce, https://www.drupal.org/project/editor_file and https://www.drupal.org/project/editor_advanced_link etc. to achieve that.

The https://www.drupal.org/project/editor_file module specifically seems interesting:
> This module allows to add a button in the rich text editor active toolbar to directly upload and link files into your content. Without this module, the writer would have to upload the files on the webserver via a file field or a FTP connexion then manually create the link.

Wondering if the UX and features provided by that module make more sense and are more straight forward than what we currently have 🤔 

Once the module is enabled, it adds a "File" button to the list of available buttons for CKE:
![image](https://github.com/backdrop/backdrop-issues/assets/2423362/38824a91-73bd-4408-99a4-c9b4fd9b5f89)

Once you drag the button to the active toolbar buttons, the "File" vertical tab appears, where there's a checkbox to enable file uploads. Once that checkbox is ticked, the same settings as in Backdrop become available:
![image](https://github.com/backdrop/backdrop-issues/assets/2423362/254d59be-5f6a-4a7e-a594-b688dad7602e)

So it seems that we have all the required functionality in Backdrop already. The only missing piece in the puzzle is the "File" button. Lets add one.

**Comment by klonos** [comment]
@irinaz ^^

**Comment by irinaz** [comment]
I will do more testing, at this point I do not see button "file" in list of available buttons when I enable file uploads
<img width="1431" alt="Screenshot 2024-05-17 at 9 46 25 AM" src="https://github.com/backdrop/backdrop-issues/assets/2497005/11e6fd55-f195-4ae8-a9f5-cb325a174751">

**Comment by klonos** [comment]
Yes @irinaz, the file button in CKEditor is a feature that is not currently available for Backdrop. This issue here is a task to add that in (if there is consensus that we need to do it in core that is).

I'm curious though, in #6552 you said this:
> My ideal use case would be to have Media library button that integrates image library with documents tab in a way similar to wordpress and drupal.

Wondering how you add that functionality in Drupal core. How is it done with the Drupal sites that you have as a reference? Trying to figure out how the same workflow works in Drupal, because AFAIK you need something like https://www.drupal.org/project/editor_file or https://www.drupal.org/project/media_entity_browser or https://www.drupal.org/project/file_browser etc.

**Comment by indigoxela** [comment]
A little off topic here, as it's not for CKEditor (and shameless self-promotion, too..). A new contrib project provides that functionality for an alternative editor: https://backdropcms.org/project/tinymce_filelink

No idea, if anything of the code could get re-used (or be an inspiration) for a CKE5 plugin. The APIs of the two editors differ quite a bit.
I'd assume, that a comparable CKE5 plugin requires a lot, a looooot more code (and work). :wink:
